### PR TITLE
[terra-date-time-picker]Single field label SR announce

### PR DESCRIPTION
**Pull request by ashishkcerner**
### Summary

Field label was not reading on date-time-picker component .

**What was changed:**
Added new props for dateTimepicker `airalLabel` .

**Why it was changed:**

### Testing

This change was tested using:

- [ ] WDIO
- [x] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-10072 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-date-picker/src/DateInput.jsx [hunk 1]
```diff
@@ -30,6 +30,14 @@ import styles from './DatePicker.module.scss';
 const cx = classNames.bind(styles);
 
 const propTypes = {
+  /**
+   * An identifier used by assistive technologies like screen readers to briefly describe this Date input to users.
+   * The label is not rendered visually.
+   *
+   * **BEST PRACTICE FOR ACCESSIBILITY**: you _SHOULD_ set this to match whatever visible label you give in your UI.
```
Comment (saket2403) [inline packages/terra-date-picker/src/DateInput.jsx:37]:
```suggestion
   * * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue):Should be set to match the visible label in the UI.
```

File: packages/terra-date-picker/src/DateInput.jsx [hunk 2]
```diff
@@ -30,6 +30,14 @@ import styles from './DatePicker.module.scss';
...
+   * Currently this is optional for passivity reasons, but it will become mandatory in a future major version.
+   */
+  a11yLabel: PropTypes.string,
```
Comment (saket2403) [inline packages/terra-date-picker/src/DateInput.jsx:40]: Can we make use of the existing ariaLabel prop here?

File: packages/terra-date-picker/src/DatePicker.jsx [hunk 3]
```diff
@@ -13,6 +13,14 @@ import styles from './DatePicker.module.scss';
...
 
 const propTypes = {
+  /**
+   * An identifier used by assistive technologies like screen readers to briefly describe this Date input to users.
+   * The label is not rendered visually.
+   *
+   * **BEST PRACTICE FOR ACCESSIBILITY**: you _SHOULD_ set this to match whatever visible label you give in your UI.
+   * Currently this is optional for passivity reasons, but it will become mandatory in a future major version.
+   */
+  a11yLabel: PropTypes.string,
```
Comment (saket2403) [inline packages/terra-date-picker/src/DatePicker.jsx:23]: Can we make use of the existing ariaLabel Prop here?

File: packages/terra-date-time-picker/src/DateTimePicker.jsx [hunk 4]
```diff
@@ -14,6 +14,11 @@ import TimeClarification from './_TimeClarification';
 const cx = classNames.bind(styles);
 
 const propTypes = {
+  /**
+   * String that labels the current element. 'aria-label' must be present for accessibility.
+   */
+  ariaLabel: PropTypes.string,
```
Comment (saket2403) [inline packages/terra-date-time-picker/src/DateTimePicker.jsx:20]: Description can be more detailed like the one you have added in date input and datePicker

File: packages/terra-date-time-picker/src/DateTimePicker.jsx [hunk 5]
```diff
@@ -778,6 +784,7 @@ class DateTimePicker extends React.Component {
 
         <div className={cx('date-facade')}>
           <DatePicker
+            a11yLabel={ariaLabel}
```
Comment (saket2403) [inline packages/terra-date-time-picker/src/DateTimePicker.jsx:787]: There is an ariaLabel Prop in Date Picker new prop should not be needed.

File: packages/terra-time-input/src/TimeInput.jsx [hunk 6]
```diff
@@ -18,6 +18,10 @@ import TimeSpacer from './_TimeSpacer';
 const cx = classNamesBind.bind(styles);
 
 const propTypes = {
+  /**
+   * String that labels the current element. 'aria-label' must be present for accessibility.
+   */
+  ariaLabel: PropTypes.string,
```
Comment (saket2403) [inline packages/terra-time-input/src/TimeInput.jsx:24]: This prop is not required you can pass value for exisiting `a11yLabel` from dateTimepicker.

**Comment by saket2403** [comment]
Please update the jest test files to pass ariaLabel to the date time picker and also add a wdio test.

File: packages/terra-date-picker/src/DateInput.jsx (hunk 2 above)
Comment (ashishkcerner) [reply packages/terra-date-picker/src/DateInput.jsx]: done https://github.com/cerner/terra-framework/pull/2031/commits/8d4603715fe10fee9e460c580b7f7c9635d09565

File: packages/terra-date-time-picker/src/DateTimePicker.jsx (hunk 4 above)
Comment (ashishkcerner) [reply packages/terra-date-time-picker/src/DateTimePicker.jsx]: done https://github.com/cerner/terra-framework/pull/2031/commits/8d4603715fe10fee9e460c580b7f7c9635d09565

File: packages/terra-date-time-picker/src/DateTimePicker.jsx (hunk 5 above)
Comment (ashishkcerner) [reply packages/terra-date-time-picker/src/DateTimePicker.jsx]: done https://github.com/cerner/terra-framework/pull/2031/commits/2d63b33d6e0de4b213da02c9b075de1cf5a96ae9

File: packages/terra-time-input/src/TimeInput.jsx (hunk 6 above)
Comment (ashishkcerner) [reply packages/terra-time-input/src/TimeInput.jsx]: done https://github.com/cerner/terra-framework/pull/2031/commits/2d63b33d6e0de4b213da02c9b075de1cf5a96ae9

**Comment by ashishkcerner** [comment]
> Please update the jest test files to pass ariaLabel to the date time picker and also add a wdio test.

Have updated the jest test but for wdio test no changes been made on Visual level .https://github.com/cerner/terra-framework/pull/2031/commits/2d63b33d6e0de4b213da02c9b075de1cf5a96ae9

File: packages/terra-date-time-picker/CHANGELOG.md [hunk 7]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed
+  * Added the `ariaLabel` props for SR announce.
```
Comment (sugan2416) [inline packages/terra-date-time-picker/CHANGELOG.md:6]:
```suggestion
  * Added the `ariaLabel` prop for SR announcement.
```

File: packages/terra-date-picker/src/DateInput.jsx [hunk 8]
```diff
@@ -732,7 +732,7 @@ const DatePickerInput = (props) => {
       size="2"
       pattern="\d*"
       aria-required={required}
-      aria-label={intl.formatMessage({ id: 'Terra.datePicker.dayLabel' })}
+      aria-label={`${ariaLabel || ''} ${intl.formatMessage({ id: 'Terra.datePicker.dayLabel' })}`}
```
Comment (sugan2416) [inline packages/terra-date-picker/src/DateInput.jsx:735]: Since `ariaLabel` is being used in https://github.com/cerner/terra-framework/blob/main/packages/terra-date-picker/src/DateInput.jsx#L853C178-L853C183 . Can you re-validate SR if there's any repeated announcement in both SR

File: packages/terra-date-picker/src/DateInput.jsx [hunk 9]
```diff
@@ -763,7 +763,7 @@ const DatePickerInput = (props) => {
       size="2"
       pattern="\d*"
       aria-required={required}
-      aria-label={intl.formatMessage({ id: 'Terra.datePicker.monthLabel' })}
+      aria-label={`${ariaLabel || ''} ${intl.formatMessage({ id: 'Terra.datePicker.monthLabel' })}`}
```
Comment (sugan2416) [inline packages/terra-date-picker/src/DateInput.jsx:766]: Can we move up this condition to remove the extra space seen in jest snaps https://github.com/cerner/terra-framework/blob/8d4603715fe10fee9e460c580b7f7c9635d09565/packages/terra-date-picker/tests/jest/__snapshots__/DatePicker.test.jsx.snap#L550

File: packages/terra-date-picker/src/DateInput.jsx (hunk 8 above)
Comment (ashishkcerner) [reply packages/terra-date-picker/src/DateInput.jsx:735]: Yes Sugan, Have tested this scenario few times. It is not being read repeatedly.

File: packages/terra-time-input/src/_AccessibleInput.jsx [hunk 10]
```diff
@@ -6,6 +6,10 @@ import { v4 as uuidv4 } from 'uuid';
 import TimeUtil from './TimeUtil';
 
 const propTypes = {
+  /**
+   * String that labels the current element. 'aria-label' must be present for accessibility.
+   */
+  ariaLabel: PropTypes.string,
```
Comment (sugan2416) [inline packages/terra-time-input/src/_AccessibleInput.jsx:12]: why is this added? TimeInput is currently not passing this

File: packages/terra-date-picker/src/DateInput.jsx (hunk 9 above)
Comment (ashishkcerner) [reply packages/terra-date-picker/src/DateInput.jsx:766]: This condition has been added to handle the scenario when user doesn't pass the ariaLabel in such case SR reads as undefined if this condition has removed .

Comment (sugan2416) [reply packages/terra-date-picker/src/DateInput.jsx:766]: If it doesnt get passed, then aria-label should be only set to the intl string directly. This will remove that single space.  Condition can be moved out to handle this

Comment (ashishkcerner) [reply packages/terra-date-picker/src/DateInput.jsx]: Yes Right have made the change https://github.com/cerner/terra-framework/pull/2031/commits/dbacc7e8f6ee54df11ab377b0bf50949bf093e3e

File: packages/terra-time-input/src/_AccessibleInput.jsx (hunk 10 above)
Comment (ashishkcerner) [reply packages/terra-time-input/src/_AccessibleInput.jsx]: Yes thanks, have made the change https://github.com/cerner/terra-framework/pull/2031/commits/dbacc7e8f6ee54df11ab377b0bf50949bf093e3e

**Review by saket2403 — APPROVED** [review]

**Review by sugan2416 — APPROVED** [review]

**Comment by rbsree** [comment]
The input fields present for Date and Time in the Terra Date Time Picker component should be grouped inside <fieldset> tag with the label "Enter Date/Time" implemented inside <legend> tag.

**Comment by ashishkcerner** [comment]
Created the jira for above https://jira2.cerner.com/browse/UXPLATFORM-10229
